src/Util.ts

```diff
@@ -267,3 +267,27 @@ function setAnchor(list: any[], predicate: PagingPredicate, nearestPage: number)
         predicate.setAnchor(nearestPage, anchor);
     }
 }
+
+export class Task {
+    intervalId: NodeJS.Timer;
+    timeoutId: NodeJS.Timer;
+}
+
+export function scheduleWithRepetition(callback: (...args: any[]) => void, initialDelay: number,
+                                       periodMillis: number): Task {
+    const task = new Task();
+    task.timeoutId = setTimeout(function (): void {
+        callback();
+        task.intervalId = setInterval(callback, periodMillis);
+    }, initialDelay);
+
+    return task;
+}
+
+export function cancelRepetitionTask(task: Task): void {
```

Comment:
what happens if I cancel the same task twice?

Reply:
I am not sure but why do we cancel same task twice ?

Reply:
i can imagine a scenario that there are two different code paths leading to cancel a task. In that case, the second one to reach cancellation code, should not cause a runtime error.

Reply:
I added a test for that scenario. There is no error.